mls-rs/src/external_client/group.rs

```diff
@@ -83,12 +85,12 @@ use alloc::boxed::Box;
 
 /// The result of processing an [ExternalGroup](ExternalGroup) message using
 /// [process_incoming_message](ExternalGroup::process_incoming_message)
-#[derive(Clone, Debug)]
+#[cfg_attr(all(feature = "ffi", not(test)), safer_ffi_gen::ffi_type(opaque))]
 #[allow(clippy::large_enum_variant)]
-pub enum ExternalReceivedMessage {
-    /// State update as the result of a successful commit.
+pub enum ExternalReceivedMessage<'a, C: ExternalClientConfig> {
```

Comment:
Just food for thought, the 'a nature of this might get in the way of passing this around or storing it places. Maybe it makes more sense for the oneshot function to return simply ExternalReceivedMessage, and the non-oneshot function to return an enum that is either ExternalReceivedMessage or CommitProcessor